x/cronos/keeper/evmhandlers/send_to_ibc_v2.go

```diff
@@ -99,8 +99,9 @@ func (h SendToIbcV2Handler) Handle(
 	}
 
 	contractAddr := sdk.AccAddress(contract.Bytes())
-	sender := sdk.AccAddress(topics[1].Bytes())
-	recipient := string(topics[2].Bytes())
+	// needs to crope the extra bytes in the topic by using BytesToAddress
+	sender := sdk.AccAddress(common.BytesToAddress(topics[1].Bytes()).Bytes())
```

Comment:
address should cast to common.Address?

Reply:
the ibc sender should be a cosmos address I think

Reply:
I still have that reflex from [v1](https://github.com/crypto-org-chain/cronos/blob/2d5fca8bec5a8a969e2b77c7746d3dc3d6183e66/x/cronos/keeper/evmhandlers/send_to_ibc.go#L88), but seem could to make version [extendable](https://github.com/mmsqe/cronos/commit/2acfc0894016efed73aac2c97d98a4797157cbf2)